integ/src/main.rs

```
@@ -0,0 +1,274 @@
+use log::info;
+use snafu::ResultExt;
+use std::env::temp_dir;
+use std::fs;
+use std::{env, process};
+use tokio::time::{sleep, Duration};
+
+use integ::ec2_provider::{create_ec2_instance, terminate_ec2_instance};
+use integ::eks_provider::{get_cluster_info, write_kubeconfig};
+use integ::error::ProviderError;
+use integ::updater::{delete_cluster_resources, label_node, run_brupop};
+
+type Result<T> = std::result::Result<T, error::Error>;
+
+/// The default path for kubeconfig file
+const KUBECONFIG_FILE_NAME: &str = "kubeconfig.yaml";
+
+/// The default region for the cluster.
+const DEFAULT_REGION: &str = "us-west-2";
+const CLUSTER_NAME: &str = "brupop-integration-test";
+
+/// This value configure how long it sleeps between create instance and label instance.
+const INTEGRATION_TEST_DELAY: Duration = Duration::from_secs(60);
+
+/// Stores global arguments.
+#[derive(Debug)]
+struct Args {
+    cluster_name: String,
+    region: String,
+    kube_config_path: String,
+}
+impl Default for Args {
+    fn default() -> Self {
+        Self {
+            cluster_name: CLUSTER_NAME.to_string(),
+            region: DEFAULT_REGION.to_string(),
+            kube_config_path: match temp_dir().to_str() {
```

Base on @cbgbt comment, better to default `kube_config_path` here. However, `temp_dir` will return `option`, and `default` function and `parse_args` can't handle `option`. Therefore, I have to manually handle if option happens here that manually generate a string path. If you have better idea, please help me on improving this. thanks!

---

I think this would happen if the path is Non-UTF-8. That seems like a reasonably obscure case such that you could probably error out instead of assuming where the temp dir is.

```rust
temp_dir().to_str().context("Unable to find temp directory")?
```

---

Yeah, I was trying to use the `context`; however, the default function didn't allow me to because it was not return option or result. Therefore,  `.context("Unable to find temp directory")?` can't be used there. Instead of, I add a new logic to compute the kube config path when `kube_config_path ` doesn't have input value. 
```
async fn generate_kube_config_path(args_kube_config_path: &str) -> Result<String> {
    let kube_config_path = match args_kube_config_path {
        DEFAULT_KUBECONFIG_FILE_NAME => format!(
            "{}/{}",
            temp_dir().to_str().context(error::FindTmpDir)?,
            DEFAULT_KUBECONFIG_FILE_NAME
        ),
        res => res.to_string(),
    };

    Ok(kube_config_path)

}
```